unstructured_ingest/v2/processes/connectors/kafka.py

```diff
@@ -0,0 +1,197 @@
+    pass
+
+
+@dataclass
+class KafkaIndexer(Indexer):
+    connection_config: KafkaConnectionConfig
+    index_config: KafkaIndexerConfig
+    connector_type: str = CONNECTOR_TYPE
+
+    def _get_messages(self):
+        from confluent_kafka import KafkaError
+
+        consumer = self.connection_config.get_client()
+        running = True
+
+        collected = {}
+        num_messages_to_consume = self.connection_config.num_messages_to_consume
+        logger.info(f"config set for blocking on {num_messages_to_consume} messages")
+        # Consume specified number of messages
+        while running:
+            msg = consumer.poll(timeout=self.connection_config.timeout)
+            if msg is None:
+                logger.debug("No Kafka messages found")
+                continue
+            if msg.error():
+                if msg.error().code() == KafkaError._PARTITION_EOF:
+                    # End of partition event
+                    logger.error(
+                        "%% %s [%d] reached end at offset %d\n"
+                        % (msg.topic(), msg.partition(), msg.offset())
+                    )
+            else:
+                msg_content = json.loads(msg.value().decode("utf8"))
+                collected[
+                    f"{msg.topic()}_{msg.partition()}_{msg.offset()}_{msg_content['filename']}"
+                ] = msg_content
+                logger.debug(f"found {len(collected)} messages, stopping")
+                consumer.commit(asynchronous=False)
+                break
+
+        return collected
+
+    def run(self) -> Generator[FileData, None, None]:
+        messages_consumed = self._get_messages()
+        for key, value in messages_consumed.items():
+            yield FileData(
+                identifier=key.split("_")[0],
+                connector_type=self.connector_type,
+                metadata=FileDataSourceMetadata(
+                    date_processed=str(time()),
+                ),
+                additional_metadata={
+                    "filename": value["filename"],
+                    "content": value["content"],
+                },
+            )
+
+    async def run_async(self, file_data: FileData, **kwargs: Any) -> download_responses:
+        raise NotImplementedError()
+
+    def precheck(self):
+        try:
+            _ = self.connection_config.get_client()
+        except Exception as e:
+            logger.error(f"failed to validate connection: {e}", exc_info=True)
+            raise SourceConnectionError(f"failed to validate connection: {e}")
+
+
+class KafkaDownloaderConfig(DownloaderConfig):
+    pass
+
+
+@dataclass
+class KafkaDownloader(Downloader):
+    connection_config: KafkaConnectionConfig
+    download_config: KafkaDownloaderConfig = field(default_factory=lambda: KafkaDownloaderConfig)
+    connector_type: str = CONNECTOR_TYPE
+    version: Optional[str] = None
+    source_url: Optional[str] = None
+
+    def _tmp_download_file(self, filename: str):
```

Comment:
A new approach taken with `v2` is specifying the filepath and filename using the `FileData` source identifiers: [example](https://github.com/Unstructured-IO/unstructured-ingest/blob/main/unstructured_ingest/v2/processes/connectors/fsspec/fsspec.py#L173). Can we populate that with the filepath and use it here when downloading.